libraries/lib-wave-track/WaveTrack.cpp

```diff
@@ -2000,14 +2000,16 @@ void WaveTrack::PasteOne(
         for (const auto& clip : track.mClips)
             if (clip->GetPlayStartTime() > t0 - (1.0 / rate))
                 clip->ShiftBy(insertDuration);
-    } else if (split)
+    }
+    else
     {
-       track.SplitAt(t0);
+       if (!merge)
+          track.SplitAt(t0);
        const auto clipAtT0 = track.GetClipAtTime(t0);
        const auto t = clipAtT0 ? clipAtT0->GetPlayEndTime() : t0;
        if (!track.IsEmpty(t, t + insertDuration))
           throw notEnoughSpaceException;
-       if (clipAtT0)
+       if (clipAtT0 && clipAtT0->GetPlayStartTime() == t0)
```

Comment:
Same question here as in the previous commit:  should times be rounded before comparison?

Reply:
They probably should. Addressing ...